src/cdomains/apron/apronDomain.apron.ml

```diff
@@ -8,6 +8,46 @@ open GobApron
 open RelationDomain
 open SharedFunctions
 
+
+(* Relational anaylsis implementation selection *)
+open ApronImplementation
+open ElinaImplementation
+
+module type Implementation =
+sig
+  (* Oct *)
+  type t
+  val to_oct : 'a Apron.Abstract1.t -> t Apron.Abstract1.t
+  val of_oct : t Apron.Abstract1.t -> 'a Apron.Abstract1.t
+  val widening_thresholds : t Apron.Manager.t -> t Apron.Abstract0.t -> t Apron.Abstract0.t -> Apron.Scalar.t array -> t Apron.Abstract0.t
+  val manager_is_oct : 'a Apron.Manager.t -> bool
+  val manager_to_oct : 'a Apron.Manager.t -> t Apron.Manager.t
+  val substitute_texpr_with : 'a Apron.Manager.t -> 'a Apron.Abstract1.t -> Apron.Var.t -> Apron.Texpr1.t -> 'a Apron.Abstract1.t option -> unit
+  val narrowing : t Apron.Manager.t -> t Apron.Abstract0.t -> t Apron.Abstract0.t -> t Apron.Abstract0.t
+  val manager_alloc : unit -> t Apron.Manager.t
+  (* Poly *)
+  type pt
+  val manager_alloc_loose : unit -> pt Apron.Manager.t
+  (* Other *)
+  val hash : 'a Apron.Manager.t -> 'a Apron.Abstract1.t -> int
+  val impl : unit -> string
+  val bound_texpr : 'a Manager.t -> string -> 'a Abstract1.t -> Texpr1.t -> Interval.t
+end
+
+let implementation impl =
+  lazy (
```

Comment:
Is there a benefit of using `lazy` here when `implementation` is a function? Also having an argument of type `string` is more error prone than using a specific variant type

Reply:
The function has been removed in favor of inlining the modules where needed.